Makefile

```diff
@@ -111,6 +111,10 @@ ifeq ($(GOLANGCI_LINT),)
 endif
 	golangci-lint run --timeout 5m --go 1.23 ./...
 
+.PHONY: check
+check: go-mod-download fmt vet golangci-lint ## Run checks for code quality and best practices
```

Comment:
I don't think that `make check` is required since `make generate` will do the work.